integration-tests/api/__tests__/admin/fulfillment-sets.spec.ts

```diff
@@ -0,0 +1,170 @@
+                postal_expression: { type: "regex", exp: "00*" },
+              },
+            ],
+          },
+          adminHeaders
+        )
+
+        const fset = response.data.fulfillment_set
+
+        expect(response.status).toEqual(200)
+        expect(fset).toEqual(
+          expect.objectContaining({
+            name: "Fulfillment Set",
+            type: "shipping",
+            service_zones: expect.arrayContaining([
+              expect.objectContaining({
+                name: "Test Zone",
+                fulfillment_set_id: fulfillmentSetId,
+                geo_zones: expect.arrayContaining([
+                  expect.objectContaining({
+                    country_code: "dk",
+                    type: "country",
+                  }),
+                  expect.objectContaining({
+                    country_code: "fr",
+                    type: "province",
+                    province_code: "fr-idf",
+                  }),
+                  expect.objectContaining({
+                    country_code: "it",
+                    type: "city",
+                    city: "some city",
+                    province_code: "some-province",
+                  }),
+                  expect.objectContaining({
+                    country_code: "it",
+                    type: "zip",
+                    city: "some city",
+                    province_code: "some-province",
+                    postal_expression: { type: "regex", exp: "00*" },
+                  }),
+                ]),
+              }),
+            ]),
+          })
+        )
+      })
+
+      it("should throw if invalid type is passed", async () => {
+        const stockLocationResponse = await api.post(
+          `/admin/stock-locations`,
+          {
+            name: "test location",
+          },
+          adminHeaders
+        )
+
+        const stockLocationId = stockLocationResponse.data.stock_location.id
+
+        const locationWithFSetResponse = await api.post(
+          `/admin/stock-locations/${stockLocationId}/fulfillment-sets?fields=id,*fulfillment_sets`,
+          {
+            name: "Fulfillment Set",
+            type: "shipping",
+          },
+          adminHeaders
+        )
+
+        const fulfillmentSetId =
+          locationWithFSetResponse.data.stock_location.fulfillment_sets[0].id
+
+        const errorResponse = await api
+          .post(
+            `/admin/fulfillment-sets/${fulfillmentSetId}/service-zones`,
+            {
+              name: "Test Zone",
+              geo_zones: [
+                {
+                  country_code: "dk",
+                  region_name: "SJAELLAND",
+                  type: "region",
+                },
+              ],
+            },
+            adminHeaders
+          )
+          .catch((err) => err.response)
+
+        expect(errorResponse.status).toEqual(400)
+        expect(errorResponse.data.message).toContain("Invalid request body: ")
```

Comment:
suggestion: I believe it would be nice to stringify the fields errors such that if someone by mistake change the validation on the server, the tests will fail, you can look how I did it here https://github.com/medusajs/medusa/pull/6977/files#diff-a625ad0732b6b0e619f1d04bdb38d7a9387a6f385ea022e69608fa8589ed86fdR81

Reply:
Nice, I'll do this 👍